pandas/core/generic.py

```diff
@@ -6296,44 +6296,44 @@ def __getattr__(self, name: str):
     @final
     def __setattr__(self, name: str, value) -> None:
         """
-        After regular attribute access, try setting the name
+        If it can be found nowhere else than the info_axis, it must be a column.
         This allows simpler access to columns for interactive use.
         """
-        # first try regular attribute access via __getattribute__, so that
-        # e.g. ``obj.x`` and ``obj.x = 4`` will always reference/modify
-        # the same attribute.
-
-        try:
-            object.__getattribute__(self, name)
-            return object.__setattr__(self, name, value)
-        except AttributeError:
-            pass
-
-        # if this fails, go on to more involved attribute setting
-        # (note that this matches __getattr__, above).
-        if name in self._internal_names_set:
-            object.__setattr__(self, name, value)
-        elif name in self._metadata:
-            object.__setattr__(self, name, value)
-        else:
+        if (
+            name not in self.__dict__
+            and name not in self._internal_names_set
+            and name not in self._metadata
+            and name in self._info_axis
+        ):
             try:
-                existing = getattr(self, name)
-                if isinstance(existing, Index):
-                    object.__setattr__(self, name, value)
-                elif name in self._info_axis:
-                    self[name] = value
-                else:
-                    object.__setattr__(self, name, value)
+                self[name] = value
             except (AttributeError, TypeError):
-                if isinstance(self, ABCDataFrame) and (is_list_like(value)):
-                    warnings.warn(
-                        "Pandas doesn't allow columns to be "
-                        "created via a new attribute name - see "
-                        "https://pandas.pydata.org/pandas-docs/"
-                        "stable/indexing.html#attribute-access",
-                        stacklevel=find_stack_level(),
-                    )
-                object.__setattr__(self, name, value)
+                pass
+            else:
+                return
+
+        if (
+            # only relevant with dataframes
+            isinstance(self, ABCDataFrame)
+            # ignore if attr already exists
+            and name not in self.__dict__
+            # ignore if it's a class attribute;
+            # might be a prop, axis, cached_property,
+            # or other sort of descriptor
+            and not hasattr(type(self), name)
+            # ignore if internal or metadata
```

Comment:
These comments are almost verbatim repetitions of the code; I think they can be removed.